### PR TITLE
[HW-2] Implement custom Array methods

### PR DESCRIPTION
**Pull request by Rediemer1**
[HW-2] Implement custom Array methods. Add array_each, array_select, array_map methods

### CODE REVIEW

File: HW02/Anatoliy_Smirnov/CustomArrayMethods.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+module CustomArrayMethods
+  refine Array do
+    def array_each
+     for l_cnt in 0...size
+       yield(self[l_cnt])
+     end
+    end
+
+    def array_select
+      l_new_array = []
```
Comment (B0RlS) [inline HW02/Anatoliy_Smirnov/CustomArrayMethods.rb:11]: Why do you need those strange letters `l` at the beginning?

Comment (Rediemer1) [reply HW02/Anatoliy_Smirnov/CustomArrayMethods.rb:11]: Hi. It's a habit to use l_% format for local variables. I’m Oracle db developer and Oracle recommend to use l_% for local variables. There was no name convention document 😉 That is why I used it for this task. If necessary I can change them, no problem

Comment (B0RlS) [reply HW02/Anatoliy_Smirnov/CustomArrayMethods.rb:11]: This is very good, but I don’t think this is necessary in the ruby code.
For example l_cnt is just a bunch of letters for me now.

Comment (Rediemer1) [reply HW02/Anatoliy_Smirnov/CustomArrayMethods.rb:11]: Yes, I understand that it is Ruby :) But the habit... you know :D

Comment (Rediemer1) [reply HW02/Anatoliy_Smirnov/CustomArrayMethods.rb:11]: For future tasks I will make in the way you require.

Comment (B0RlS) [reply HW02/Anatoliy_Smirnov/CustomArrayMethods.rb:11]: Cool, thanks

**Pull request merged**